iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/rest/v1/handler/StatementConstructionHandler.java

```diff
@@ -164,4 +172,55 @@ public static InsertTabletStatement constructInsertTabletStatement(
     insertStatement.setAligned(insertTabletRequest.getIsAligned());
     return insertStatement;
   }
+
+  public static InsertRowsStatement createInsertRowsStatement(
+      InsertRecordsRequest insertRecordsRequest)
+      throws IllegalPathException, QueryProcessException, IoTDBConnectionException {
+
+    // construct insert statement
+    InsertRowsStatement insertStatement = new InsertRowsStatement();
+    List<InsertRowStatement> insertRowStatementList = new ArrayList<>();
+    List<List<TSDataType>> dataTypesList = new ArrayList<>();
+
+    for (int i = 0; i < insertRecordsRequest.getDataTypesList().size(); i++) {
+      List<TSDataType> dataTypes = new ArrayList<>();
+      for (int c = 0; c < insertRecordsRequest.getDataTypesList().get(i).size(); c++) {
+        dataTypes.add(
+            TSDataType.valueOf(
+                insertRecordsRequest.getDataTypesList().get(i).get(c).toUpperCase(Locale.ROOT)));
+      }
+      dataTypesList.add(dataTypes);
+    }
+
+    InsertRowDataUtils.filterNullValueAndMeasurement(
+        insertRecordsRequest.getDeviceIds(),
+        insertRecordsRequest.getTimestamps(),
+        insertRecordsRequest.getMeasurementsList(),
+        insertRecordsRequest.getValuesList(),
+        dataTypesList);
+
+    List<ByteBuffer> valuesList =
+        InsertRowDataUtils.objectValuesListToByteBufferList(
+            insertRecordsRequest.getValuesList(), dataTypesList);
+
+    for (int i = 0; i < insertRecordsRequest.getDeviceIds().size(); i++) {
+      InsertRowStatement statement = new InsertRowStatement();
+      statement.setDevicePath(
+          DataNodeDevicePathCache.getInstance()
+              .getPartialPath(insertRecordsRequest.getDeviceIds().get(i)));
+      statement.setMeasurements(
+          insertRecordsRequest.getMeasurementsList().get(i).toArray(new String[0]));
+      statement.setTime(insertRecordsRequest.getTimestamps().get(i));
+      statement.fillValues(valuesList.get(i));
```

Comment:
I think convert the values to byte buffer and transfer back to objects is not a good idea. We can just set values directly to the insertStatement.